packages/snap-preact/src/Instantiators/RecommendationInstantiator.tsx

```diff
@@ -135,6 +139,23 @@ export class RecommendationInstantiator {
 				if (options?.limit && Number.isInteger(Number(options?.limit))) {
 					contextGlobals.limits = Number(options?.limit);
 				}
+				let cartContents;
+				if (typeof cart === 'function') {
+					try {
+						const cartFuncContents = cart();
+						if (Array.isArray(cartFuncContents)) {
+							cartContents = cartFuncContents;
+						}
+					} catch (e) {
+						this.logger.warn(`Error getting cart contents from function`, e);
+					}
+				} else if (Array.isArray(cart)) {
+					cartContents = cart;
+				}
+				if (Array.isArray(cartContents)) {
+					(this.config.services?.tracker || this.tracker).cookies.cart.set(cartContents);
```

Comment:
I think we ought to be setting `this.tracker` (along with other services) to the config services (if provided) during construction.

```
(this.config.services?.tracker || this.tracker).cookies.cart.set(cartContents);
```
Should become:
```
this.tracker.cookies.cart.set(cartContents);
```

It looks like we are doing this for services throughout the instantiator - but perhaps it is a good time to refactor and make this better.